### ISSUE TITLE
Chat glyphs collide after joining enough channels

### ISSUE THREAD

**Issue by lycurgus** [urbit/urbit#3114]
**Describe the bug**
I'm a member of 28 chats. There are 27 unique glyphs available for applying to chats. As a result, two of my chats have the same glyph.

**To Reproduce**
1. Join 28 chats
2. Issue a `;what` in chat-cli
3. See that two chats have the same glyph

**Expected behavior**
Glyphs should not be applied to chats if all the available glyphs are already used. Alternatively, glyphs could roll over to digraphs once the first 27 are expended.

**Screenshots**
![image](https://user-images.githubusercontent.com/8443742/87010040-57b20480-c209-11ea-877b-96316f15ee84.png)

**Desktop (please complete the following information):**
 - Base hash of your urbit ship: 0vc.5mb6v.nqu78.n190s.qdfcr.5o28s.kcf0q.nuem2.9d4p8.06sp7.dhk61

**Additional context**
I have a suspicion that this is related to #3113 somehow as I think the large unread counts may have started after I joined the 28th channel.

**Comment by Fang-** [comment]
Yeah, chat-cli's glyphs kind of lose their luster once you start being in many chats. Then, I guess that holds true for chat-cli in general.

This can't be related to #3113, as chat-cli doesn't touch unread counts, and is just a client to chat-store, just like landscape is.

Do you have a proposal for expanding the space of available glyphs? If not, I'd be tempted to close this as "wontfix until we fix cli apps in general", because the _real_ solution here is to not display everything in a single stream all of the time.

**Comment by lycurgus** [comment]
My suspicion of the relation to 3113 was rather that _Landscape_ was somehow getting confused by the duplicate glyphs and potentially misattributing something - but I suppose if it was something like that, I'd need to have thousands of unreads against some channel for it to get wrong, whereas it seems to be creating them out of thin air at present... Not to mention that I don't imagine Landscape's chat even looks at the glyphs.

As far as a proposal for glyph space expansion, my only real thought would be going to digraphs as mentioned - once the single-glyph space is exhausted, move on to `!@`, `!#`, and so on. That takes the space north of 700 channels before running out again. I guess a problem there is it changes the spacing of chat-cli

I'd be on board with deferring this to be addressed by a broader change to not display everything in a single stream, as you mentioned - my concern there is that display is only one half of the issue. Even if the chat histories aren't shown intermingled (and therefore don't need a glyph to distinguish the audience of each line) you still need some way to easily choose a target for sending from the cli. Unless chat-cli gets retired altogether in favour of the web version? Which also makes sense given that other Landscape apps don't have a cli equivalent...

**Comment by Fang-** [comment]
> you still need some way to easily choose a target for sending from the cli.

It's only half a solution, but fwiw using a glyph bound to multiple targets select the most recently active target.

https://github.com/urbit/urbit/blob/69f8602e55c9467f6a8b714af5cd9ecd7224eb04/pkg/arvo/app/chat-cli.hoon#L421

> Unless chat-cli gets retired altogether in favour of the web version? Which also makes sense given that other Landscape apps don't have a cli equivalent...

There might be a command line renaissance on the horizon. No timeline yet, but I'm personally excited about giving people the cli apps they deserve.
